src/NServiceBus.Core/MessageInterfaces/MessageMapper/Reflection/MessageMapper.cs

```diff
@@ -65,7 +65,8 @@ public Type GetMappedTypeFor(Type t)
                 return Type.GetTypeFromHandle(typeHandle);
             }
 
-            return null;
+            InitType(t);
+            return GetMappedTypeFor(t);
```

Comment:
I wonder if this could turn into a infinite loop? perhaps a private method with a "allowTypeBeAdded" flag?